### PR TITLE
planner/core: migrate test-infra to testify for preprocess_test.go

### LINKED ISSUE #28395: migrate test-infra to testify for `planner/core/preprocess_test.go`

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #28395

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/preprocess_test.go [hunk 1]
```diff
@@ -304,42 +273,52 @@ func (s *testValidatorSuite) TestValidator(c *C) {
...
+	require.NoError(t, err)
 
-	_, err = s.se.Execute(context.Background(), "create database test2")
-	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "create table test.t2(d int)")
+	require.NoError(t, err)
 
-	_, err = s.se.Execute(context.Background(), "create table test2.t(e int)")
-	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "create database test2")
```
Comment (hawkingrei) [inline planner/core/preprocess_test.go:303]: use ```tk.MustExec```.

ditto other.

Comment (tisonkun) [reply planner/core/preprocess_test.go]: Updated.

**Review by hawkingrei — APPROVED** [review]

**Review by mjonss — APPROVED** [review]
LGTM

**Pull request merged**